### DISCUSSION TITLE
Problem installing two fake google services (location & gcam)

### DISCUSSION THREAD

**Discussion by VeritasVeritus** [lukaspieper/Gcam-Services-Provider#19] [General]
I'm running a custom rom with no gapps and with [UnifiedNlp](https://github.com/microg/UnifiedNlp) Now I'm trying to install Gcam Services Provider. When trying to install the latter, I get "App not installed" and I'm guessing this is because both fake services use com.google.android.gms. Is there any solution to this without installing microG or other form of gapps?

`adb shell pm install --force-queryable /data/local/tmp/app-debug.apk`
`Failure [INSTALL_FAILED_VERSION_DOWNGRADE]`

`adb shell pm install --force-queryable -d /data/local/tmp/app-debug.apk`
`Failure [INSTALL_FAILED_UPDATE_INCOMPATIBLE: Package com.google.android.gms signatures do not match previously installed version; ignoring!]`

Thank you for your work.

**Comment by lukaspieper** [comment]
Hi,

you can change the PackageName, it doesn't have to be `com.google.android.gms`. But I chose this PackageName to avoid possible conflicts with microg or GApps (Play services).

I didn't know your use case when I made this decision, but seems reasonable to me and could affect others. 

Can you download and try the apk from [here](https://github.com/lukaspieper/Gcam-Services-Provider/actions/runs/463682733) (Artifacts => App)?

**Reply by VeritasVeritus** [reply]
> you can change the PackageName, it doesn't have to be com.google.android.gms.

This solved it, if inexperienced users as me wants to do this, open the source with android studio, find in path android.gms, replace the results with the package name you want and build the apk. Gcam is now working!

**Reply by ghost** [reply]
> Hi,
> 
> you can change the PackageName, it doesn't have to be `com.google.android.gms`. But I chose this PackageName to avoid possible conflicts with microg or GApps (Play services).
> 
> I didn't know your use case when I made this decision, but seems reasonable to me and could affect others.
> 
> Can you download and try the apk from [here](https://github.com/lukaspieper/Gcam-Services-Provider/actions/runs/463682733) (Artifacts => App)?

Bruh apk link expired :-( reupload pls? 🙏

**Reply by lukaspieper** [reply]
Hi,

your problem is already solved. Please have a look [here](https://github.com/lukaspieper/Gcam-Services-Provider/discussions/19#discussioncomment-1272897).

**Comment by VeritasVeritus** [comment]
I tried installing the apk you linked from artifacts and had the same issue. I also opened it with android studio, I think the package name is still the same:
```
    xmlns:android="http://schemas.android.com/apk/res/android"
    android:versionCode="2"
    android:versionName="1.1"
    android:compileSdkVersion="30"
    android:compileSdkVersionCodename="11"
    package="com.google.android.gms"
    platformBuildVersionCode="30"
    platformBuildVersionName="11"
```


I'm a beginner, maybe I've missed something.
Thank you for you help.

**Reply by lukaspieper** [reply]
Sry, my bad.

I just updated the branch ([artifact](https://github.com/lukaspieper/Gcam-Services-Provider/actions/runs/463792591)).

**Reply by Dreomite** [reply]
@lukaspieper
This solved the issue. Now I can use UnifiedNLP and your app at the same time. Thank you!

**Comment by ghost** [comment]
Hi Lukas.

The artifact expired.
Could you create a new one please?

Thanks for this tool. :)

**Reply by lukaspieper** [reply]
Hi, new artifact is available [here](https://github.com/lukaspieper/Gcam-Services-Provider/actions/runs/690751805).

Will probably use the changed package name permanently from a next release.

**Reply by ghost** [reply]
Thanks a lot.

Package name permanently changed sounds good.

**Reply by mmowchenko** [reply]
I just wanted to say I'm having the same issue. Much thanks to all for figuring this out!

**Comment by lukaspieper** [comment]
Finally I created a [new release](https://github.com/lukaspieper/Gcam-Services-Provider/releases/tag/v1.2) with the new package name. Downloading from artifacts should no longer be needed.

**Reply by ghost** [reply]
Cool !

Thanks.

**Reply by ghost** [reply]
> Cool !
> 
> Thanks.

@lukaspieper  app is not installed 🗿🗿🗿🗿🗿🗿🗿🗿🗿🗿

**Comment by rotonott** [comment]
@lukaspieper 

```sh
$ adb install app-debug.apk 
Performing Streamed Install
adb: failed to install app-debug.apk: Failure [INSTALL_FAILED_CONFLICTING_PROVIDER: Package couldn't be installed in /data/app/de.lukaspieper.gcam.services-DkVAwrbXb0Y3Qy6wMJqQ_g==: Can't install because provider name com.google.android.gsf.gservices (in package de.lukaspieper.gcam.services) is already used by com.google.android.gsf]
```

What can be done in this situation?

stock Android 9 (unrooted)
LH-H930DS (LG V30)

**Reply by lukaspieper** [reply]
Hi,

first of all thanks for researching your error message and not just creating a new issue 👍🏻 

When you're running stock Android, I assume that you have Google Play Services built-in, so the conflict you get is expected and correct, that's the way the Android operating system works. 

So the question here is, what do you want to achieve? What problem are you facing and hoping to resolve by installing one of the app flavors of this project?

**Reply by lukaspieper** [reply]
Btw you can install the APKs of this project regularly without adb, there was a time you had to use adb but that problem was resolved.

Furthermore there hasn't been an APK that contained the world "debug" for several years now.

**Reply by rotonott** [reply]
Hi, the reason I posted my issue here is that I am facing the same issue as the original poster (`Failure [INSTALL_FAILED_UPDATE_INCOMPATIBLE: Package com.google.android.gms signatures do not match previously installed version; ignoring!]`) when trying to install the latest apk. I followed your advice from
> Finally I created a [new release](https://github.com/lukaspieper/Gcam-Services-Provider/releases/tag/v1.2) with the new package name. Downloading from artifacts should no longer be needed.

and tried to install that apk, to no avail. 

So I have debloated and degoogled to the largest extent my phone would let me (including removing the gsf and gms packages with [uad-ng](https://github.com/Universal-Debloater-Alliance/universal-android-debloater-next-generation)). But I need Gcam-Services-Provider [to use the Google Camera port app](https://www.celsoazevedo.com/files/android/google-camera/troubleshooting/#help12) so I can replace the low quality stock Camera app. I really prefer if I can get away with not having spyware (google services) on my phone.

Thanks a lot for your prompt response, I really appreciate it

**Reply by rotonott** [reply]
Just a quick note, it's night for me, so I might respond in 8-10 hours to new messages. Thanks!

**Reply by lukaspieper** [reply]
> Thanks a lot for your prompt response, I really appreciate it
> Just a quick note, it's night for me, so I might respond in 8-10 hours to new messages. Thanks!

Don't worry. Yeah, you just got lucky with my quick response. It's definitely not the norm 😆 

> I posted my issue here is that I am facing the same issue as the original poster (Failure [INSTALL_FAILED_UPDATE_INCOMPATIBLE: Package com.google.android.gms signatures do not match previously installed version; ignoring!]) 

That's generally good practice, and it does fit here somehow, but it's actually not the same issue. Don't worry about it, though. The OP faced a signature conflict of the packages (apps). Your error message shows a conflict of the `com.google.android.gsf.gservices` provider because every provider can only exist in one app on your whole system.

> I followed your advice from "Finally I created a [new release](https://github.com/lukaspieper/Gcam-Services-Provider/releases/tag/v1.2) with the new package name. Downloading from artifacts should no longer be needed." and tried to install that apk, to no avail.

You have to consider, that post is from 2021 and it references version 1.2 of this project. The current version is [1.6.1](https://github.com/lukaspieper/Gcam-Services-Provider/releases). But a newer version won't fix the "error" you get because the installer wouldn't allow installation anyway.

> So I have debloated and degoogled to the largest extent my phone would let me (including removing the gsf and gms packages with [uad-ng](https://github.com/Universal-Debloater-Alliance/universal-android-debloater-next-generation)).

And here we get to the core issue. The problem is that these tools are not capable of **removing** system apps. The good thing about the tool you chose is that it actually admits it [here](https://github.com/Universal-Debloater-Alliance/universal-android-debloater-next-generation/wiki/FAQ#how-is-it-possible-to-delete-a-system-package-without-root-permission). Therefore, the Google Play Services still conflict with my project, and the installer correctly prevents the installation.

I'm sorry but there is nothing I can do from my side.
